### DISCUSSION TITLE
Why can't the publisher and subscriber communicate after the participant is configured with my_listener?

### DISCUSSION THREAD

**Discussion by MAXIANH0NG** [eProsima/Fast-DDS#4449] [Q&A] [answered]
Here are two different ways to create participants:
//m_Native = DomainParticipantFactory::get_instance()->create_participant(domainId, PARTICIPANT_QOS_DEFAULT, &my_listener);
m_Native = DomainParticipantFactory::get_instance()->create_participant(domainId, PARTICIPANT_QOS_DEFAULT);

That is to say, once I configure my_listener for the participant, the on_data_available inside my SubListener will not be called. Why is that? I have tried not configuring my_listener for the participant, and indeed, on_data_available is called. Could it be that the two are incompatible?
![be67822be2fb63c777825e60719a9e8](https://github.com/eProsima/Fast-DDS/assets/136770138/a3321b74-3ffc-4406-a7ec-99c5006c96d7)

**Comment by Mario-DL** [comment, marked as answer]
Hi @MAXIANH0NG 
Thanks for using `Fast DDS`. Please, have a look at the `StatusMask` documentation [here](https://fast-dds.docs.eprosima.com/en/latest/fastdds/dds_layer/core/entity/entity.html?highlight=on_data_on_readers#id1). It sounds to me that the top participant listener is otherwise shadowing the inner `SubListener` call.

**Reply by MAXIANH0NG** [reply]
It worked, thank you so much, I really want to bow down to you, this problem has troubled me for several days. 
![9d11d714f0d9c141b8aac17fd039f8c](https://github.com/eProsima/Fast-DDS/assets/136770138/37106a34-a709-4d8a-a49c-1894d5b96930)

```
        StatusMask mask = StatusMask::all();
        mask &= ~StatusMask::data_on_readers();
        m_Native = DomainParticipantFactory::get_instance()->create_participant(domainId, pqos, &my_listener, mask);
```
